### ISSUE TITLE
foo in <template #foo> returns a TemplateRef not a TemplateElement

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular_analyzer_plugin#367]
```
<template #propertyListItemTemplate let-listItem="item">
```

with 

```
    <list-component 
           [template]="propertyListItemTemplate"
```

and

```dart
class ListComponent {
  @Input()
  TemplateRef template;
```

causes 
```
warning: Attribute value expression (of type TemplateElement) is not assignable to component input (of type TemplateRef) (input_binding_type_error at [fhir_designer] lib/webui/component_property_editor/component_property_editor_component.html:15)
```

but I think this should not cause a warning and the code is working as intened.

**Comment by MichaelRFairhurst** [comment]
Yes, I do believe you're right @zoechi.

@maxkim can you take a look? We have the TemplateRef ClassElement available
via standardAngular.templateRef

**Comment by zoechi** [comment]
In TS land they split the functionality. They have now `<ng-template>` for above use case and `<template>` is a plain old `TemplateElement`.
Perhaps it's time to get that into Dart Angular as well.
It limits for example using Polymer with Angular.

**Comment by zoechi** [comment]
@matanlurey any plans for `<ng-template>` in Dart Angular?

**Comment by mk13** [comment]
Taking a look.

**Comment by mk13** [comment]
Resolved - please raise another issue if a related issue arises.
